geomstats/geometry/connection.py

```diff
@@ -230,29 +279,32 @@ def pole_ladder_parallel_transport(
           ⟨hal-02148832⟩
         """
         current_point = gs.copy(base_point)
-        transported_tangent_vector = gs.copy(tangent_vec_a)
+        transported_tangent_vec = gs.copy(tangent_vec_a)
         base_shoot = self.exp(base_point=current_point,
-                              tangent_vec=transported_tangent_vector)
+                              tangent_vec=transported_tangent_vec)
+        trajectory = []
         for i_point in range(0, n_steps):
             frac_tangent_vector_b = (i_point + 1) / n_steps * tangent_vec_b
             next_point = self.exp(
                 base_point=base_point,
                 tangent_vec=frac_tangent_vector_b)
-            transported_tangent_vector, base_shoot = self.pole_ladder_step(
+            transported_tangent_vec, base_shoot, pts = self._pole_ladder_step(
                 base_point=current_point,
                 next_point=next_point,
-                base_shoot=base_shoot)
+                base_shoot=base_shoot,
+                **single_step_kwargs)
             current_point = next_point
+            trajectory.append(pts)
 
-        return transported_tangent_vector
+        return transported_tangent_vec, trajectory
```

Comment:
Same question here

Reply:
Yes, same probleme:
```
pole_ladder, trajectory = metric.ladder_parallel_transport(
    tan_a, tan_b, point, n_steps=n_steps, return_geodesics=True)
```
would run, but
```
pole_ladder, trajectory = metric.ladder_parallel_transport(
    tan_a, tan_b, point, n_steps=n_steps, return_geodesics=False)
```
would crash, which I think is more annoying than always returning a tuple even if one may be an empty list